crowbar_framework/app/models/batch/export.rb

```diff
@@ -0,0 +1,177 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Batch
+  class Export < Base
+    def path
+      @path ||= Rails.root.join(
+        "storage",
+        "batch",
+        "export",
+        filename
+      )
+
+      unless @path.dirname.directory?
+        @path.dirname.mkpath
+      end
+
+      @path
+    end
+
+    def filename
+      @filename ||= begin
+        time = Time.now.utc.strftime(
+          "%Y%m%d-%H%M%S%z"
+        )
+
+        "batch-#{time}.yml"
+      end
+    end
+
+    protected
+
+    def persist!
+      path.open(File::CREAT | File::TRUNC | File::RDWR) do |file|
+        file.write content.to_yaml
+      end
+
+      true
+    end
+
+    def content
+      struct = [].tap do |result|
+        barclamps.each do |barclamp|
+          next unless process_barclamp?(
+            barclamp
+          )
+
+          value = process_barclamp!(
+            barclamp
+          )
+
+          result.concat(
+            value
+          ) if value.present?
+        end
+      end
+
+      {
+        "proposals" => struct
+      }
+    end
+
+    def process_barclamp?(barclamp)
+      if excluder = expanded_includes.present? \
+        && expanded_includes.keys.exclude?(barclamp)
+        return false
+      end
+
+      if expanded_excludes.present? \
+        && expanded_excludes.keys.include?(barclamp)
+        return false
+      end
+
+      true
+    end
+
+    def process_barclamp!(barclamp)
+      [].tap do |result|
+        Proposal.where(
+          barclamp: barclamp
+        ).each do |proposal|
+          next unless process_proposal?(
+            proposal
+          )
+
+          value = process_proposal!(
+            proposal
+          )
+
+          result.push(
+            value
+          ) if value.present?
+        end
+      end
+    end
+
+    def process_proposal?(proposal)
+      if expanded_includes.present? \
+        && expanded_includes[proposal.barclamp].present? \
+        && expanded_includes[proposal.barclamp].exclude?(proposal.name)
+        return false
+      end
+
+      if expanded_excludes.present? \
+        && expanded_excludes[proposal.barclamp].present? \
+        && expanded_excludes[proposal.barclamp].include?(proposal.name)
+        return false
+      end
+
+      true
+    end
+
+    def process_proposal!(proposal)
+      {}.tap do |result|
+        result["barclamp"] = proposal.barclamp
+
+        unless proposal.name == "default"
+          result["name"] = proposal.name
+        end
+
+        removed, added = template(proposal.barclamp).easy_diff(
+          proposal.raw_attributes.to_hash
+        )
+
+        to_wipe = squash(removed) - squash(added)
+
+        unless to_wipe.empty?
+          result["wipe_attributes"] = to_wipe
+        end
+
+        if added.empty?
```

Comment:
Style/ConditionalAssignment: Use the return of the conditional for variable assignment and comparison.